File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Reports/Commands/ReportTaskResumeCommand.js
SIE.defineCommand('SIE.Web.MES.TaskManagement.Reports.Commands.ReportTaskResumeCommand', {
    extend: 'SIE.cmd.Command',
    meta: { text: "恢复", group: "edit", iconCls: "icon-Reload icon-blue" },
    executeIntervalMode: SIE.cmd.IntervalMode.Debounce.value,
    canExecute: function (view) {
        var selecteditems = view.getSelection();
        if (selecteditems != null && selecteditems.length > 0) {
            for (var i = 0; i < selecteditems.length; i++) {
                if (selecteditems[i].data.TaskStatus !== 40) {
                    return false;
                }
            }
            return true;
        }
        return false;
    },
    execute: function (view, source) {
        SIE.Msg.wait("正在恢复......".t());
        view.execute({
            withIds: true,
            selectIds: view.getSelectionIds(),
            success: function (res) { //回调
                var errMsg = res.Result;
                if (errMsg == '恢复成功'.t())
                    view.reloadData();
                SIE.Msg.showMessage(errMsg);
            }
        });
    }
});

File: Modules/SMES/Tasks/SIE.Web.MES.TaskManagement/Reports/Commands/ReportTaskResumeCommand.cs
using SIE.MES.TaskManagement.Dispatchs;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIE.Web.MES.TaskManagement.Reports.Commands
{
    /// <summary>
    /// 报工关联恢复命令
    /// </summary>
    public class ReportTaskResumeCommand : ViewCommand
    {
        /// <summary>
        /// 执行恢复操作
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>恢复结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            string errMsg = string.Empty;
            var selectedIds = args.SelectedIds.ToList();
            errMsg = RT.Service.Resolve<DispatchController>().SetResumeTasks(selectedIds);
            if (errMsg.Length == 0)
                return "恢复成功".L10N();
            else
                return errMsg;
        }
    }
}
